ols/src/query_helpers/docs_summarizer.py

```diff
@@ -16,9 +18,36 @@
 logger = logging.getLogger(__name__)
 
 
+class _NoLLMMetadataPostprocessor(BaseNodePostprocessor):
+    """A LlamaIndex PostProcessor that prevents node metadata from reaching the LLM."""
+
+    def _postprocess_nodes(
+        self, nodes: list[NodeWithScore], query_bundle: Optional[QueryBundle] = None
+    ) -> list[NodeWithScore]:
+        for n in nodes:
+            n.node.excluded_llm_metadata_keys = list(n.node.metadata.keys())
+        return nodes
+
+
 class DocsSummarizer(QueryHelper):
     """A class for summarizing documentation context."""
 
+    @staticmethod
+    def _file_path_to_doc_url(file_path: str) -> str:
+        """Convert file_path metadata to the corresponding URL on the OCP docs website.
+
+        Embedding node metadata 'file_path' in the form
+        file_path: /workspace/source/ocp-product-docs-plaintext/hardware_enablement/
+                    psap-node-feature-discovery-operator.txt
+        is mapped into a doc URL such as
+        https://docs.openshift.com/container-platform/4.14/hardware_enablement/psap-node-feature-discovery-operator.html.
```

Comment:
i expect this code to be temporary and that very soon we fix our RAG embedding pipeline to construct this URL and bake it into the metadata of the RAG content itself.

cc @ilan-pinto

Reply:
@ilan-pinto There's a link per embeddings document, there can be many of them if a query pulls > 1 document